MetaMask/metamask-extension
# Performance improvements for Swaps

## Explanation
This PR improves performance in Swaps, especially on the Build Quote page (~7.2x faster!)

I've used 2 main tools for performance improvements:
- https://github.com/welldone-software/why-did-you-render
- Performance tab in Google Chrome

## Screenshots
Before: Total Time: 3390 ms to render the BuildQuote component with token lists with 4x CPU slowdown
![image](https://user-images.githubusercontent.com/80175477/181248008-561544fa-d4a1-4795-b26d-fcdaae43e9c1.png)

After: Total Time: 469 ms to render the BuildQuote component with token lists with 4x CPU slowdown
![image](https://user-images.githubusercontent.com/80175477/181482800-bb65fef2-1ce5-4856-ace1-d4ac7c720b4e.png)


## Manual Testing Steps
- Run Swaps and see if everything is running smoothly now
- In Chrome's dev tools you can switch to the Performance tab and record a Build Quote page reload with 4x CPU slowdown (with and without this PR to compare results)
